### DISCUSSION TITLE
The dock feature

### DISCUSSION THREAD

**Discussion by petros** [zed-industries/zed#6474] [Show and tell]
I think this may be a less known feature. You can have a dock anchored to the right or bottom of the editor. It's often useful for keeping your terminal sessions in there. But you can have all sorts of files open too. The nice thing is that it remembers what's open even if you quit and reopen Zed.

You can make the dock appear using <kbd>CMD</kbd>+<kbd>Shift</kbd>+<kbd>k</kbd>+<kbd>Down</kbd> or <kbd>Right</kbd> to anchor to the bottom or right. Or type `dock` in the command palette (<kbd>CMD</kbd>+<kbd>Shift</kbd>+<kbd>p</kbd>) to check all options.

Here's a short video demo:

https://user-images.githubusercontent.com/28818/218049167-62506149-74db-4d1c-ab81-11d1c6172fd6.mp4

**Comment by JosephTLyons** [comment]
Awesome tip!

**Comment by exalted** [comment]
Hmm, nice! But that begs the question: What's the difference between a pane and a dock? And I don't mean it from a technical perspective but rather from a UX point of view.

It seems to me that the dock works like a pane but can also be resized, so why have both and not "upgrade" everything to dock and get rid of panes?

**Reply by exalted** [reply]
Oh, I see! You can only have one dock, but panes can be many! Is that it? If so, seems a good 'ol feature merge of these two (i.e., dock and panes) seems reasonable.

**Reply by petros** [reply]
Yes, that's one aspect of it.

I am also trying to understand the differences myself 😃. Let me share my current understanding.

The dock is special in that it contains a pane. The dock gives special functionality to that pane. It makes it hide-able, you can anchor it to the right or at the bottom, or hover it on top of the rest of the UI. And, at least now, there's only one dock. And of course, you can resize it.

One might say, we could have added that functionality to all panes. But I can see why we want to separate this from a UI perspective. Not all current or future characteristics may make sense for all panes.

Having said that, I cannot exclude a redesign in the future. 😃

I am not super satisfied with my answer as I am probably missing more details. But as I learn more about this, I'll share more 😄.

Your feedback is not only appreciated, but please keep it coming 😄

**Reply by JosephTLyons** [reply]
We do also plan to introduce resizable *normal* panes as well.  It just hasn't happened yet.  I guess there's some technical difficulty surrounding doing that for that specific code at this time, which is why it hasn't happened yet.

**Reply by exalted** [reply]
Thanks for jumping in, both of you! ❤️ Still, I am puzzled and not sure quite get the _why_ these are two distinct features with somewhat overlapping features/purposes.

**Comment by diktomat** [comment]
One thing that annoys me slightly is opening a new file when the focus is currently on the Dock. A new tab will get opened in the Dock, which I / which is considered a kinda special pane. I'd prefer (a setting) for this new tab to be opened in the last-focused “regular” pane instead. If I want a file in the Dock, I'll move it there.

**Reply by petros** [reply]
It seems this is coming Soon™ 😄.

**Reply by JosephTLyons** [reply]
The new mechanism won't have a setting.  It will always load every newly-opened file into the center pane area (aka, the non-dock area).  There will be a key combo that can be used to quickly toggle an item between the dock and the center pane area.  The new workflow will be as such.

**Comment by bchamagne** [comment]
Hey, looks like the tips on the video is not valid anymore? I was looking for a way to have the terminal from the bottom dock to the right dock.

**Reply by petros** [reply]
Yes, the whole pane and docking system has been reworked since I made that video. What you are looking should be doable, but probably using a different approach.
